crates/ethereum/engine/src/orchestrator.rs

```diff
@@ -0,0 +1,224 @@
+
+        let PipelineContainer { pipeline, pipeline_task_spawner } = pipeline_container;
+        let backfill_sync = PipelineSync::new(pipeline, pipeline_task_spawner);
+
+        Ok(EthChainOrchestrator { orchestrator: ChainOrchestrator::new(handler, backfill_sync) })
+    }
+}
+
+/// The type that drives the Ethereum chain forward and communicates progress.
+#[pin_project]
+#[allow(missing_debug_implementations)]
+pub struct EthChainOrchestrator<DB, Client>
+where
+    DB: Database + 'static,
+    Client: HeadersClient + BodiesClient + Clone + Unpin + 'static,
+{
+    orchestrator: EthChainOrchestratorType<DB, Client>,
+}
+
+impl<DB, Client> Future for EthChainOrchestrator<DB, Client>
+where
+    DB: Database + 'static,
+    Client: HeadersClient + BodiesClient + Clone + Unpin + 'static,
+{
+    type Output = Result<(), EthChainOrchestratorError>;
+
+    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        // Call poll on the inner orchestrator.
+        match self.project().orchestrator.poll_next_unpin(cx) {
+            Poll::Ready(Some(_event)) => Poll::Ready(Ok(())),
+            Poll::Ready(None) => Poll::Ready(Ok(())),
+            Poll::Pending => Poll::Pending,
+        }
+    }
+}
+
+/// Potential error returned by `EthChainOrchestrator`.
+#[derive(Debug)]
+pub struct EthChainOrchestratorError {}
+
+/// Helper container for tree channels in `EthChainOrchestratorBuilder`.
+struct TreeChannels<T: EngineTypes> {
+    to_tree: mpsc::Sender<FromEngine<BeaconEngineMessage<T>>>,
+    from_tree: mpsc::UnboundedReceiver<EngineApiEvent>,
+}
+
+/// Helper container for pipeline related values in `EthChainOrchestratorBuilder`.
+struct PipelineContainer<DB: Database> {
+    pipeline: Pipeline<DB>,
+    pipeline_task_spawner: Box<dyn TaskSpawner>,
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use reth_chainspec::{ChainSpecBuilder, MAINNET};
+    use reth_engine_tree::test_utils::TestPipelineBuilder;
+    use reth_ethereum_engine_primitives::EthEngineTypes;
+    use reth_network_p2p::test_utils::TestFullBlockClient;
+    use reth_tasks::TokioTaskExecutor;
+    use std::sync::Arc;
+
+    #[test]
+    fn eth_chain_orchestrator_build() {
+        let chain_spec = Arc::new(
+            ChainSpecBuilder::default()
+                .chain(MAINNET.chain)
+                .genesis(MAINNET.genesis.clone())
+                .paris_activated()
+                .build(),
+        );
+
+        let client = TestFullBlockClient::default();
+
+        let (_tx, rx) = mpsc::unbounded_channel::<BeaconEngineMessage<EthEngineTypes>>();
+        let incoming_requests = UnboundedReceiverStream::new(rx);
+
+        let pipeline = TestPipelineBuilder::new().build(chain_spec.clone());
+        let pipeline_task_spawner = Box::<TokioTaskExecutor>::default();
+
+        let (to_tree_tx, _to_tree_rx) = mpsc::channel(32);
+        let (_from_tree_tx, from_tree_rx) = mpsc::unbounded_channel();
```

Comment:
I see, I believe what's still missing here is actually spawning the tree, which should probably happen in `EthOrchestrator::new`

Reply:
good point, currently the orchestrator doesn't include any tree-related component, should we add it now?  `EngineApiRequestHandler` is missing a db controller here https://github.com/paradigmxyz/reth/blob/main/crates/engine/tree/src/engine.rs#L152 and looking at current the instantiation of the node here https://github.com/paradigmxyz/reth/blob/main/crates/node/builder/src/launch/common.rs#L630-L645 we use the `blockchain_tree` to store a `blockchain_db`, should we do the same here?

Reply:
> tree-related component, should we add it now?

I think we can, we should however use the new types @rkrasiuk is working on and start investigating how to bootstrap the persistence and tree services. this we can do in a next step